ci_framework/roles/edpm_deploy_baremetal/tasks/main.yml

```diff
@@ -30,19 +30,25 @@
 
 - name: Override uefi image url when we build uefi image via content_provider
   when: cifmw_build_images_output is defined
-  set_fact:
-    cifmw_edpm_deploy_baremetal_uefi_image_url: "{{ cifmw_build_images_output['images']['edpm-hardened-uefi']['image'] }}"
+  block:
+    - name: set facts for cifmw_edpm_deploy_baremetal_uefi_image_url
+      ansible.builtin.set_fact:
```

Comment:
We need to override the image in periodic jobs as well, so that's why we originally kept the defaults and only update when the image is != default.